doc/nrf/gs_recommended_versions.rst

```diff
@@ -0,0 +1,187 @@
+           - |cmake_recommended_ver_win10|
+         * - dtc
+           - |dtc_min_ver|
+           - |dtc_recommended_ver_win10|
+         * - git
+           -
+           - |git_recommended_ver_win10|
+         * - GNU Arm Embedded Toolchain
+           - |gnuarmemb_min_ver|
+           - |gnuarmemb_recommended_ver_win10|
+         * - gperf
+           - |gperf_min_ver|
+           - |gperf_recommended_ver_win10|
+         * - ninja
+           - |ninja_min_ver|
+           - |ninja_recommended_ver_win10|
+         * - Python
+           - |python_min_ver|
+           - |python_recommended_ver_win10|
+         * - SEGGER Embedded Studio (Nordic Edition)
+           - |ses_min_ver|
+           - |ses_recommended_ver_win10|
+
+   .. group-tab:: Linux
+
+      .. list-table::
+         :header-rows: 1
+
+         * - Tool
+           - Minimum version
+           - Tested version
+         * - ccache
+           -
+           - |ccache_recommended_ver_linux|
+         * - CMake
+           - |cmake_min_ver|
+           - |cmake_recommended_ver_linux|
+         * - dfu_util
+           -
+           - |dfu_util_recommended_ver_linux|
+         * - dtc
+           - |dtc_min_ver|
+           - |dtc_recommended_ver_linux|
+         * - git
+           -
+           - |git_recommended_ver_linux|
+         * - GNU Arm Embedded Toolchain
+           - |gnuarmemb_min_ver|
+           - |gnuarmemb_recommended_ver_linux|
+         * - gperf
+           - |gperf_min_ver|
+           - |gperf_recommended_ver_linux|
+         * - ninja
+           - |ninja_min_ver|
+           - |ninja_recommended_ver_linux|
+         * - Python
+           - |python_min_ver|
+           - |python_recommended_ver_linux|
+         * - SEGGER Embedded Studio (Nordic Edition)
+           - |ses_min_ver|
+           - |ses_recommended_ver_linux|
+
+   .. group-tab:: macOS
+
+      .. list-table::
+         :header-rows: 1
+
+         * - Tool
+           - Minimum version
+           - Toolchain manager version
+         * - CMake
+           - |cmake_min_ver|
+           - |cmake_recommended_ver_darwin|
+         * - dtc
+           - |dtc_min_ver|
+           - |dtc_recommended_ver_darwin|
+         * - git
+           -
+           - |git_recommended_ver_darwin|
+         * - GNU Arm Embedded Toolchain
+           - |gnuarmemb_min_ver|
+           - |gnuarmemb_recommended_ver_darwin|
+         * - gperf
+           - |gperf_min_ver|
+           - |gperf_recommended_ver_darwin|
+         * - ninja
+           - |ninja_min_ver|
+           - |ninja_recommended_ver_darwin|
+         * - Python
+           - |python_min_ver|
+           - |python_recommended_ver_darwin|
+         * - SEGGER Embedded Studio (Nordic Edition)
+           - |ses_min_ver|
+           - |ses_recommended_ver_darwin|
+
+
+Required Python dependencies
+############################
+
+The following table shows the Python packages that are required for working with |NCS| |version|.
```

Comment:
Is this actually true? Or does it show **some of** the Python packages (namely the ones that are in the list below)?

Reply:
it should show all packages required to build and run Zephyr, mcuboot, and nrf, as we parse all three requirements.txt.

And by your request, also the required doc packages.

What it doesn't show is the ncs extras that are used for upmerge handling, and ci tools, that simply defines specific version, as those are not needed for the above building and running.

Reply:
also extra packages in Zephyr are not shown, such as gitlint, lpc_checksum, gcovr, etc. as those are not strictly needed for building and running (but if following the getting started, then users do get those installed)

Reply:
And if someone happens a new tool to the requirements.txt, but not to the list?
This WILL happen.

Reply:
That was what I thought when I asked how the `.txt` files are updated. We should have a single source of truth regarding versions, be it one or more files, requirements.txt or whatever, that *all* tooling and *all* build systems, CI, etc agree to use, but I don't think it is something in the scope of this PR.

Reply:
> We should have a single source of truth regarding versions, be it one or more files, requirements.txt or whatever

and that truth is exactly the `requirements*.txt` and `versions*.txt` files.

What @ru-fu is asking for, is getting the tables in the docs generated based on those files, and not only the version information.
And that part is outside the scope of this PR.